### PR TITLE
Adding env variable for easily switching data contexts.

### PR DESCRIPTION
**Pull request by merivale**
Soon Marlowe Run will be working with the PAB. For now, we're using localStorage to simulate persistent and shared data to aid with testing. But even when we're working with the PAB, there are two options: (1) the plain PAB with the Marlowe Plutus apps available as separte executables; (2) the "Marlowe PAB", i.e. a single executable of the PAB with the Marlowe Plutus apps bundled up with it. Depending on which of these we're using, we activate Marlowe Plutus apps differently - either with the path to the separate executable (for option 1), or with a type that the bundled executable understands (for option 2).

I want to make it as easy as possible to switch between these three different options (the two PABs and localStorage). So this PR adds a new `DataProvider` env variable, which is read by the `ManageMarlowe` capability. To change options, all you have to do is change the env variable, and voila: the `ManageMarlowe` capability takes care of the rest.

I've also updated the `ContractHistory` type to match changes recently made to the follower contract.

Oh, and I've changed the PAB port to 8080 (only affects local development), because the marlowe-pab doesn't read the plutus-pab.yaml file and is hard-coded to port 8080. Jann says the marlowe-pab will be able to read the config file soon.

I've tested this with the regular plutus-pab and with the marlowe-pab, and the basics seem to be working. I couldn't test the plutus-pab very thoroughly, because wallets appear to have no initial funds to play with at the moment.

### CODE REVIEW

File: marlowe-dashboard-client/src/API/Contract.purs [hunk 1]
```diff
@@ -13,18 +14,27 @@ import API.Request (doGetRequest, doPostRequest, doPutRequest)
...
 import Effect.Aff.Class (class MonadAff)
-import Foreign.Generic (class Encode)
+import Foreign.Generic (class Decode, class Encode)
 import Plutus.PAB.Effects.Contract.ContractExe (ContractExe)
 import Plutus.PAB.Webserver.Types (ContractActivationArgs, ContractInstanceClientState, ContractSignatureResponse)
 import Servant.PureScript.Ajax (AjaxError)
 import Wallet.Emulator.Wallet (Wallet)
 import Wallet.Types (ContractInstanceId)
 
+-- PAB contracts can be activated either with a ContractExe (a wrapper around a path the the exe on disk), or
```
Comment (hrajchert) [inline marlowe-dashboard-client/src/API/Contract.purs:24]: small typo "the the"

**Review by hrajchert — APPROVED** [review]
LGTM

File: marlowe-dashboard-client/src/Capability/Contract.purs [hunk 2]
```diff
@@ -34,43 +34,43 @@ import WalletData.Types (Wallet)
...
-  deactivateContract :: PlutusAppId -> m (AjaxResponse Unit)
-  getContractInstanceClientState :: PlutusAppId -> m (AjaxResponse (ContractInstanceClientState ContractExe))
-  getContractInstanceCurrentState :: PlutusAppId -> m (AjaxResponse (PartiallyDecodedResponse ActiveEndpoint))
-  getContractInstanceObservableState :: PlutusAppId -> m (AjaxResponse RawJson)
-  getContractInstanceHooks :: PlutusAppId -> m (AjaxResponse (Array (Request ActiveEndpoint)))
-  invokeEndpoint :: forall d. Encode d => PlutusAppId -> String -> d -> m (AjaxResponse Unit)
-  getWalletContractInstances :: Wallet -> m (AjaxResponse (Array (ContractInstanceClientState ContractExe)))
-  getAllContractInstances :: m (AjaxResponse (Array (ContractInstanceClientState ContractExe)))
-  getContractDefinitions :: m (AjaxResponse (Array (ContractSignatureResponse ContractExe)))
+  activateContract :: forall a. ContractActivationId a => a -> Wallet -> m (AjaxResponse PlutusAppId)
```
Comment (hrajchert) [inline marlowe-dashboard-client/src/Capability/Contract.purs:37]: looking at this usage I wonder if `Id` is the noun we are looking for... And looking at the endpoint below `deactivateContract`, I wonder if the `Activation` part is correct...

Maybe `ContractExe a` is the way to go, but I'd need to understand more all the options in order to name them.

**Review by hrajchert — APPROVED** [review]
LGTM

File: marlowe-dashboard-client/src/Contract/State.purs [hunk 3]
```diff
@@ -84,9 +84,9 @@ dummyState =
   emptyMarloweState = Semantic.State { accounts: mempty, choices: mempty, boundValues: mempty, minSlot: zero }
 
 mkInitialState :: WalletDetails -> Slot -> PlutusAppId -> ContractHistory -> Maybe State
-mkInitialState walletDetails currentSlot followerAppId contractHistory = case contractHistory of
-  None -> Nothing
-  History marloweParams marloweData transactionInputs ->
+mkInitialState walletDetails currentSlot followerAppId { chParams, chHistory } = case chParams of
```
Comment (hrajchert) [inline marlowe-dashboard-client/src/Contract/State.purs:87]: chParams >>= \(marloweParams /\ marloweData)

Comment (merivale) [reply marlowe-dashboard-client/src/Contract/State.purs:87]: Good point. I think I prefer `bind` here, though I'm not quite sure why.

File: marlowe-dashboard-client/src/Capability/Contract.purs (hunk 2 above)
Comment (merivale) [reply marlowe-dashboard-client/src/Capability/Contract.purs:37]: We can't use `ContractExe`, as that's already taken. Here, as elsewhere, I'm letting the Haskell names guide me. The `ContractActivationArgs` type is a record with a `caWallet` (contract activation wallet) and a `caID` (contract activation id) - and the latter is what my type here is for. I don't think it's a bad name: the type is intended to be a way of identifying the contract you want to activate. You either do that with the path to the executable (`ContractExe`) or some other built-in type.

**Pull request merged**